elastic/kibana
# [Infrastructure UI] Asset Details clean up and improvements

Part of: [#156696](https://github.com/elastic/kibana/issues/156696)

## Summary

This PR refactors some of the newly introduced Asset Details embeddable code to make it easier to consume by other pages, decoupling its props from what the Host View used to send to the old Host View Flyout.

Besides adjustments to the component responsiveness, this PR also improves the logic around adding tabs and managing their state.

### How to test this PR

- Run `yarn storybook infra`
  - Check all the storybooks related to the Asset Detail Embeddable
- Start Kibana
  - Go to the Hosts View page, open the flyout, and play with it. 

### For reviewers
The Asset Details Embeddable still depends on a few things from the `pages/*` folder, and a couple of components are still tightly coupled with the Host View. We'll address these things to keep this PR small in the next ones.
